From fcb0bbe7cbfd8302ce7d7bfd86d02e3182f613aa Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Wed, 29 Apr 2026 14:46:29 +0800
Subject: [PATCH] dts: arm: ast2705: add EVB device tree

Add the AST2705 SoC include and EVB board device tree based on
AST2700, with AST2705 SOC1 reset and clock compatibles.

Include the new DTB in the AST2700 build list so the AST2705 EVB tree
can be built by the normal dtbs target.

Upstream-Status: Backport [SDK v11.03 migration]
Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I24e2cae617f95701107d309c6c830e5e38d37006
Signed-off-by: rupeshs <rupeshs@ami.com>
---
 arch/arm/dts/Makefile        |   1 +
 arch/arm/dts/ast2705-evb.dts | 318 +++++++++++++++++++++++++++++++++++
 arch/arm/dts/ast2705.dtsi    |  31 ++++
 3 files changed, 350 insertions(+)
 create mode 100644 arch/arm/dts/ast2705-evb.dts
 create mode 100644 arch/arm/dts/ast2705.dtsi

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 2632022440..29ce14ffd9 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1285,6 +1285,7 @@ dtb-$(CONFIG_ASPEED_AST2600) += \
 
 dtb-$(CONFIG_ASPEED_AST2700) += \
 	ast2700-evb.dtb \
+	ast2705-evb.dtb \
 	ast2700-raw.dtb \
 	ast2700-slt.dtb \
 	ast2700-ncsi.dtb \
diff --git a/arch/arm/dts/ast2705-evb.dts b/arch/arm/dts/ast2705-evb.dts
new file mode 100644
index 0000000000..16f53f854c
--- /dev/null
+++ b/arch/arm/dts/ast2705-evb.dts
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: GPL-2.0+
+
+/dts-v1/;
+
+#include "ast2705.dtsi"
+#include "ast2700-u-boot.dtsi"
+#include <dt-bindings/net/ti-dp83867.h>
+
+#define DUAL_NODE 0	// 1: DUAL_NODE, 0: SINGLE_NODE
+#define EDAF_128MB 0	// 1: 128MB, 0: 64MB
+
+/ {
+	model = "AST2705 EVB";
+	compatible = "aspeed,ast2700-evb", "aspeed,ast2700";
+
+	memory {
+		device_type = "memory";
+		reg = <0x4 0x00000000 0x0 0x20000000>;
+	};
+
+	chosen {
+		stdout-path = &uart12;
+	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+
+		#include "ast2700-evb-reserved-mem.dtsi"
+	};
+};
+
+&uart12 {
+	status = "okay";
+};
+
+&mdio0 {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	ethphy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+};
+
+&mdio1 {
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	ethphy1: ethernet-phy@0 {
+		reg = <0>;
+		/* For DDR5 board */
+		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+		ti,tx-internal-delay = <DP83867_RGMIIDCTL_2_00_NS>;
+	};
+};
+
+&mac0 {
+	status = "okay";
+	phy-mode = "rgmii-id";
+	phy-handle = <&ethphy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii0_default>;
+	rx-internal-delay-ps = <0>;
+	tx-internal-delay-ps = <0>;
+};
+
+&mac1 {
+	status = "okay";
+	phy-mode = "rgmii-id";
+	phy-handle = <&ethphy1>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii1_default>;
+	rx-internal-delay-ps = <0>;
+	tx-internal-delay-ps = <0>;
+};
+
+&fmc {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_fwspi_quad_default>;
+
+	flash@0 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+
+	flash@1 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+
+	flash@2 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi0_default &pinctrl_spi0_cs1_default>;
+
+	flash@0 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+
+	flash@1 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_spi1_cs1_default>;
+
+	flash@0 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+
+	flash@1 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+};
+
+&spi2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_spi2_cs1_default>;
+
+#if 1
+	flash@0 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+
+	flash@1 {
+		status = "okay";
+		spi-max-frequency = <50000000>;
+		spi-tx-bus-width = <2>;
+		spi-rx-bus-width = <2>;
+	};
+#else
+	compatible = "aspeed,ast2700-spi-txrx";
+	spi-aspeed-full-duplex;
+	tpm0: tpmdev@0 {
+		compatible = "tcg,tpm_tis-spi";
+		spi-max-frequency = <25000000>;
+		reg = <0>;
+		status = "okay";
+	};
+#endif
+};
+
+&emmc_controller {
+	status = "okay";
+	sdhci_hs200; /* enable hs200 */
+};
+
+&emmc {
+	status = "okay";
+	max-frequency = <200000000>;
+};
+
+&sdc {
+	status = "okay";
+	sdhci_hs200; /* enable sdr104 */
+
+	vcc_sdio: vcc-sdio {
+		status = "okay";
+		compatible = "regulator-fixed";
+		gpios = <&gpio0 ASPEED_GPIO(G, 6) GPIO_ACTIVE_HIGH>;
+		regulator-name = "vcc_sdio";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		enable-active-high;
+		regulator-boot-on;
+	};
+
+	vccq_sdio: vccq-sdio {
+		status = "okay";
+		compatible = "regulator-gpio";
+		gpios = <&gpio0 ASPEED_GPIO(G, 7) GPIO_ACTIVE_HIGH>;
+		gpios-states = <1>;
+		states = <1800000 0x0>,
+			 <3300000 0x1>;
+		regulator-name = "vccq_sdio";
+		regulator-type = "voltage";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+};
+
+&sdhci {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sdio_default>;
+	vmmc-supply = <&vcc_sdio>;
+	vqmmc-supply = <&vccq_sdio>;
+	/*
+	 * When running SDR104 mode, it suggests to set driving to 3.
+	 * When running HS mode, it suggests to set driving to 2.
+	 */
+	sdhci-drive-type = <3>;
+	max-frequency = <125000000>;
+};
+
+&ufs {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+};
+
+&i2c1 {
+	status = "okay";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&i2c3 {
+	status = "okay";
+};
+
+&i2c8 {
+	status = "okay";
+};
+
+&i2c11 {
+	status = "okay";
+};
+
+&wdt0 {
+	status = "okay";
+};
+
+&hace {
+	status = "okay";
+};
+
+&display_port {
+	status = "okay";
+};
+
+&vbios {
+	status = "okay";
+};
+
+&vga {
+	status = "okay";
+};
+
+&vhuba0 {
+	status = "okay";
+};
+
+&vhubb0 {
+	status = "okay";
+};
+
+&vhubc {
+	status = "okay";
+};
+
+&vhubd {
+	status = "okay";
+};
+
+#if 0
+&soc0 {
+	mbox-ssp-0 {
+		compatible = "aspeed,aspeed-mbox";
+		reg = <0x4 0x31480000 0x0 0x200000>, <0x0 0x10001000 0x0 0x1000>;
+		mboxes = <&mbox0 0>;
+	};
+};
+
+&soc1 {
+	mbox-bootmcu-0 {
+		compatible = "aspeed,aspeed-mbox";
+		reg = <0x4 0x31a80000 0x0 0x80000>, <0x4 0x31a88000 0x0 0x80000>;
+		mboxes = <&mbox2 0>;
+	};
+};
+#endif
+
+&soc0 {
+	mbox-bootmcu-1 {
+		compatible = "aspeed,aspeed-mbox";
+		reg = <0x4 0x31880000 0x0 0x100000>, <0x4 0x31980000 0x0 0x100000>;
+		mboxes = <&mbox2 1>;
+	};
+};
diff --git a/arch/arm/dts/ast2705.dtsi b/arch/arm/dts/ast2705.dtsi
new file mode 100644
index 0000000000..d5aeac423e
--- /dev/null
+++ b/arch/arm/dts/ast2705.dtsi
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0+
+#include "ast2700.dtsi"
+
+/ {
+	soc1: soc@14000000 {
+		syscon1: syscon@14c02000 {
+			compatible = "aspeed,ast2705-scu1", "syscon", "simple-mfd";
+			reg = <0x0 0x14c02000 0x0 0x1000>;
+			ranges = <0 0 0 0x14c02000 0 0x1000>;
+			#address-cells = <2>;
+			#size-cells = <2>;
+
+			soc1_rst: reset-controller@14c02200 {
+				compatible = "aspeed,ast2705-reset";
+				reg = <0 0x200 0 0x40>;
+				#reset-cells = <1>;
+			};
+
+			soc1_clk: clock-controller@14c02200 {
+				compatible = "aspeed,ast2705-soc1-clk";
+				reg = <0 0x240 0 0x1c0>;
+				#clock-cells = <1>;
+			};
+
+			pinctrl1: pinctrl@14c02400 {
+				compatible = "aspeed,ast2700-soc1-pinctrl";
+				reg = <0 0x400 0 0x100>;
+			};
+		};
+	};
+};
-- 
2.43.0

